.github/workflows/tests.yml

```diff
@@ -1,18 +1,27 @@
 name: Tests
 
 on:
-  pull_request:
-    branches: [ "*" ]
+  push:
```

Comment:
You are correct: this mean runs for every push to every branch.

It didn't run for this PR because the workflow file has to exist in the `main` branch before it can be run. In subsequent PRs, when this file already exists, it will be run for those as well.